code/game/objects/structures/crates_lockers/closets/bodybag.dm

```diff
@@ -147,3 +148,163 @@
 		max_weight_of_contents = A_is_item.w_class
 	B.w_class = max_weight_of_contents
 	usr.put_in_hands(B)
+
+/// Environmental bags
+
+/obj/structure/closet/body_bag/environmental
+	name = "environmental protection bag"
+	desc = "An insulated, reinforced bag designed to protect against exoplanetary storms and other environmental factors."
+	icon = "icons/obj/bodybag.dmi"
+	icon_state = "envirobag"
+	mob_storage_capacity = 1
+	pressure_protection = 0.8
+	thermal_insulation = 0.5
+	foldedbag_path = /obj/item/body_bag/environmental/
+	weather_protection = list("acid", "ash", "rad", "snow", "void") // Does not protect against lava or the The Floor Is Lava spell.
+
+/obj/structure/closet/body_bag/environmental/nanotrasen
+	name = "elite environmental protection bag"
+	desc = "Used for VIP extraction from hazardous environments and as shelter by elite Nanotrasen operatives, this highly reinforced bag is nearly completely isolated from environmental factors when closed."
+	icon = "icons/obj/bodybag.dmi"
+	icon_state = "ntenvirobag"
+	pressure_protection = 1
+	thermal_insulation = 1
+	foldedbag_path = /obj/item/body_bag/environmental/nanotrasen/
+	weather_protection = list("all")
+
+/// Securable enviro. bags
+
+/obj/structure/closet/body_bag/environmental/prisoner
+	name = "prisoner transport bag"
+	desc = "Intended for transport of prisoners through hazardous environments, this environmental protection bag comes with straps to keep an occupant secure."
+	icon = "icons/obj/bodybag.dmi"
+	icon_state = "prisonerenvirobag"
+	foldedbag_path = /obj/item/body_bag/environmental/prisoner/
+	breakout_time = 3000 // Five minutes, because it's probably about as hard to get out of this as it is to get out of a straightjacket.
```

Comment:
this sounds incredibly unfun
also use time defines

Reply:
link for that in contributing.md

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-our-time-defines